torch/testing/_asserts.py

```diff
@@ -102,6 +97,63 @@ def wrapper(actual: Tensor, expected: Tensor, **kwargs: Any) -> Optional[Excepti
     return wrapper
 
 
+def _check_quantization(check_tensor_values: Callable[..., Optional[Exception]]) -> Callable[..., Optional[Exception]]:
+    """Decorates continuous tensor check functions to handle quantized tensors.
+
+    If the inputs are not quantized, this decorator is a no-op.
+
+    Args:
+        check_tensor_values (Callable[..., Optional[Exception]]): Tensor check function for continuous tensors.
+
+    Returns:
+        Optional[Exception]: Return value of :attr:`check_tensors`.
+    """
+
+    @functools.wraps(check_tensor_values)
+    def wrapper(actual: Tensor, expected: Tensor, **kwargs: Any) -> Optional[Exception]:
+        if not actual.is_quantized:
+            return check_tensor_values(actual, expected, **kwargs)
+
+        if actual.qscheme() == torch.per_tensor_affine:
+            exc = check_tensor_values(torch.as_tensor(actual.q_scale()), torch.as_tensor(expected.q_scale()), **kwargs)
+            if exc:
+                return _amend_error_message(exc, "{}\n\nThe failure occurred for the quantization scale.")
+
+            exc = check_tensor_values(
+                torch.as_tensor(actual.q_zero_point()), torch.as_tensor(expected.q_zero_point()), **kwargs
+            )
+            if exc:
+                return _amend_error_message(exc, "{}\n\nThe failure occurred for the quantization zero point.")
+        elif actual.qscheme() == torch.per_channel_affine:
+            exc = check_tensor_values(actual.q_per_channel_scales(), expected.q_per_channel_scales(), **kwargs)
+            if exc:
+                return _amend_error_message(exc, "{}\n\nThe failure occurred for the quantization scales.")
+
+            exc = check_tensor_values(
+                actual.q_per_channel_zero_points(), expected.q_per_channel_zero_points(), **kwargs
+            )
+            if exc:
+                return _amend_error_message(exc, "{}\n\nThe failure occurred for the quantization zero points.")
+
+            exc = check_tensor_values(
+                torch.as_tensor(actual.q_per_channel_axis()), torch.as_tensor(expected.q_per_channel_axis()), **kwargs
+            )
+            if exc:
+                return _amend_error_message(exc, "{}\n\nThe failure occurred for the quantization per-channel axis.")
+        elif actual.qscheme() == torch.per_tensor_symmetric:
+            # TODO add meta data checking
+            pass
+        elif actual.qscheme() == torch.per_channel_symmetric:
+            # TODO add meta data checking
+            pass
```

Comment:
These schemes are missing in `self.assertEqual`. I can't find anything about the meta data in the documentation. Is there something that should be checked for equality here?

Reply:
Seems like we should require the schemes match? But let's give someone from the quantization team an opportunity to weigh in

Reply:
We require just that in `_check_attributes_equal`. See L209-L210.